include/libtorrent/storage.hpp

```diff
@@ -277,7 +277,7 @@ namespace libtorrent {
 		virtual ~storage_interface() {}
 
 		// initialized in disk_io_thread::perform_async_job
-		aux::session_settings* m_settings = nullptr;
+		aux::session_settings const* m_settings = nullptr;
```

Comment:
technically, this could be cleaned up and simplified now too, where this member could be initialized in the constructor (but that would be an API break). But it could at least be initialized by the main thread immediately after construction